### ISSUE TITLE
Product update method causes PHP notice in multi currency enabled shop with ps_categoryproducts module installed

### ISSUE THREAD

**Issue by reanim8ed** [PrestaShop/PrestaShop#14768]
**Describe the bug**
I'm making Prestashop integration with some WebService so I'm creating and updating products via code. Creating part works without any problems, but then trying to update excisting product I get PHP notice: Trying to get property of non-object. 
I have debugged and found that on product update the ps_categoryproducts module in 349 line calls getCacheId method in Module class. This method tries to add currency ID to cache array, but the context object doesnt have any currency set at this step: 
http://bit.ly/2Z096QP
http://bit.ly/2Z14k5n

This issue doesnt block update process, but running the script from console - it generates notice for every product which is updated.

**To Reproduce**
1. Install fresh PS version
2. Enable multi currency support: add localization with different currency than already installed: International->Localization (at the installation I selected Lithuania for the country, so Euro is used as default currency, here I add localization for Poland and additional currency is added): http://bit.ly/2StPvG7
3. Install latest [ps_categoryproducts](https://github.com/PrestaShop/ps_categoryproducts/releases) module version: http://bit.ly/2SqQE0U
4. Create new Product object for existing product in custom php file or in the module
5. Try to update it with update() method.

If I surround statement in Module class @ 2353 line with isset() check - the error is gone.

**Additional information**
PrestaShop version: 1.7.5.2
PHP version: 7.1

**Comment by khouloudbelguith** [comment]
Hi @reanim8ed,

Could you please provide your custom PHP file to check with it.

Thanks!

**Comment by reanim8ed** [comment]
In the root Prestashop directory create some folder e.g. 'test', put test.php here and run it with php via console:
[test.zip](https://github.com/PrestaShop/PrestaShop/files/3412003/test.zip)
http://bit.ly/2SoLLFX

Also I have updated the description how to reproduce this.
And the same notice is displayed on Prestashop 1.7.6 too

Probably the module can be updated to fix this, but I think the getCacheId method in Module class should make the checks for context params too

**Comment by khouloudbelguith** [comment]
@reanim8ed,

Thanks for your feedback.
I have the same issue with PS1.7.5.2 & PS1.7.6.0.
```
PHP Notice:  Trying to get property of non-object in /projet/QA/ps1752/classes/module/Module.php on line 2353
PHP Stack trace:
PHP   1. {main}() /projet/QA/ps1752/test/test.php:0
PHP   2. Product->update() /projet/QA/ps1752/test/test.php:17
PHP   3. HookCore::exec() /projet/QA/ps1752/classes/Product.php:643
PHP   4. HookCore::callHookOn() /projet/QA/ps1752/classes/Hook.php:903
PHP   5. HookCore::coreCallHook() /projet/QA/ps1752/classes/Hook.php:355
PHP   6. Ps_Categoryproducts->hookUpdateProduct() /projet/QA/ps1752/classes/Hook.php:966
PHP   7. Ps_Categoryproducts->clearCache() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:116
PHP   8. Ps_Categoryproducts->getInformationFromConfiguration() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:126
PHP   9. Ps_Categoryproducts->getCacheId() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:349

Notice: Trying to get property of non-object in /projet/QA/ps1752/classes/module/Module.php on line 2353

Call Stack:
    0.0001     356008   1. {main}() /projet/QA/ps1752/test/test.php:0
    3.8928   19362640   2. Product->update() /projet/QA/ps1752/test/test.php:17
    5.3473   19339616   3. HookCore::exec() /projet/QA/ps1752/classes/Product.php:643
    5.3509   19599672   4. HookCore::callHookOn() /projet/QA/ps1752/classes/Hook.php:903
    5.3509   19599720   5. HookCore::coreCallHook() /projet/QA/ps1752/classes/Hook.php:355
    5.3509   19599720   6. Ps_Categoryproducts->hookUpdateProduct() /projet/QA/ps1752/classes/Hook.php:966
    5.3509   19599720   7. Ps_Categoryproducts->clearCache() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:116
    5.3509   19599720   8. Ps_Categoryproducts->getInformationFromConfiguration() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:126
    5.3509   19608392   9. Ps_Categoryproducts->getCacheId() /projet/QA/ps1752/modules/ps_categoryproducts/ps_categoryproducts.php:349
```
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by thalassa-web** [comment]
The question is : Why is the context not initialized with the default currency ?

**Comment by Progi1984** [comment]
> The question is : Why is the context not initialized with the default currency ?

Fixed in #18859 (1.7.7.x / develop).
